src/Microsoft.AspNetCore.Http/PipeWriterAdapter.cs

```diff
@@ -0,0 +1,287 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Pipelines;
+using System.Runtime.CompilerServices;
+using System.Runtime.ExceptionServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Http
+{
+    /// <summary>
+    /// Implements PipeWriter using a base stream implementation. 
+    /// </summary>
+    public class PipeWriterAdapter : PipeWriter, IDisposable
+    {
+        private readonly int _minimumSegmentSize;
+        private readonly Stream _writingStream;
+        private int _bytesWritten;
+
+        private List<CompletedBuffer> _completedSegments;
+        private byte[] _currentSegment;
```

Comment:
Memory<byte>

Reply:
That’s much less efficient. Let’s measure the before and after though

Reply:
I'll measure the perf with and without using Memory here, then compare it with using IMemoryPool vs ArrayPool.